src/overlays/actors/ovl_En_Knight/z_en_knight.h

```diff
@@ -1,29 +1,155 @@
 
 struct EnKnight;
 
 typedef void (*EnKnightActionFunc)(struct EnKnight*, PlayState*);
 
+typedef enum {
+    /*   0 */ EN_KNIGHT_PARAM_IGOS_MAIN = 0,
+    /*  35 */ EN_KNIGHT_PARAM_OTHERS = 35,
+    /* 100 */ EN_KNIGHT_PARAM_IGOS_HEAD = 100,
+    /* 200 */ EN_KNIGHT_PARAM_KNIGHT_BODY_AFTERIMAGE = 200,
+    /* 201 */ EN_KNIGHT_PARAM_IGOS_BODY_AFTERIMAGE,
+    /* 202 */ EN_KNIGHT_PARAM_IGOS_HEAD_AFTERIMAGE
+} EnKnightParam;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_WAIT,
+    /* 1 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_BEGIN,
+    /* 2 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_PERFORM
+} EnKnightSubActionCurtainReact;
+
+typedef enum {
+    /* 0 */ KNIGHT_CLOSE_CURTAIN_ACTION_0,
+    /* 1 */ KNIGHT_CLOSE_CURTAIN_ACTION_1,
+    /* 2 */ KNIGHT_CLOSE_CURTAIN_ACTION_2
+} EnKnightCloseCurtainAction;
+
+typedef enum KnightBodyPart {
+    /*  0 */ KNIGHT_BODYPART_JAW,
+    /*  1 */ KNIGHT_BODYPART_TORSO,
+    /*  2 */ KNIGHT_BODYPART_PELVIS,
+    /*  3 */ KNIGHT_BODYPART_LEFT_UPPER_ARM,
+    /*  4 */ KNIGHT_BODYPART_LEFT_FOREARM,
+    /*  5 */ KNIGHT_BODYPART_SHIELD,
+    /*  6 */ KNIGHT_BODYPART_RIGHT_UPPER_ARM,
+    /*  7 */ KNIGHT_BODYPART_RIGHT_FOREARM,
+    /*  8 */ KNIGHT_BODYPART_SWORD,
+    /*  9 */ KNIGHT_BODYPART_LEFT_LEG_UPPER,
+    /* 10 */ KNIGHT_BODYPART_LEFT_LEG_LOWER,
+    /* 11 */ KNIGHT_BODYPART_LEFT_FOOT,
+    /* 12 */ KNIGHT_BODYPART_RIGHT_LEG_UPPER,
+    /* 13 */ KNIGHT_BODYPART_RIGHT_LEG_LOWER,
+    /* 14 */ KNIGHT_BODYPART_RIGHT_FOOT,
+    /* 15 */ KNIGHT_BODYPART_MAX
+} KnightBodyPart;
+
 typedef struct EnKnight {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ s16 unk_144;
-    /* 0x146 */ char unk_146[0x2];
-    /* 0x148 */ u8 unk_148;
-    /* 0x149 */ char unk_149[0x8];
-    /* 0x151 */ u8 unk_151;
-    /* 0x152 */ char unk_152[0x1];
-    /* 0x153 */ u8 unk_153;
-    /* 0x154 */ u8 unk_154;
-    /* 0x155 */ char unk_155[0x2CB];
+    /* 0x144 */ u16 curtainsLightRayAngle;
+    /* 0x146 */ s16 randTimer;
+    /* 0x148 */ u8 subAction;
+    /* 0x14A */ s16 actionTimers[3];
+    /* 0x150 */ UNK_TYPE1 unk_150;
+    /* 0x151 */ u8 inCurtainCutscene;
+    /* 0x152 */ u8 doBgChecks;
+    /* 0x153 */ u8 closeCurtainAction;
+    /* 0x154 */ u8 roomLightingState;
+    /* 0x156 */ s16 invincibilityTimer;
+    /* 0x158 */ s16 damageFlashTimer;
+    /* 0x15A */ s16 strafeTarget;
+    /* 0x15C */ s16 strafeMaxStep;
+    /* 0x15E */ s16 effectTimer;
+    /* 0x160 */ s16 breathAlpha;
+    /* 0x162 */ u8 isHeadless;
+    /* 0x164 */ Vec3f afterImageOffset;
+    /* 0x170 */ s16 pitchToPlayer;
+    /* 0x172 */ s16 yawToPlayer;
+    /* 0x174 */ s16 leftLegUpperRotation;
+    /* 0x176 */ s16 rightLegUpperRotation;
+    /* 0x178 */ s16 leftLegLowerRotation;
+    /* 0x17A */ s16 rightLegLowerRotation;
+    /* 0x17C */ f32 bodyAlpha;
+    /* 0x180 */ f32 shadowAlphaFactor;
+    /* 0x184 */ s16 neckPitch;
+    /* 0x186 */ s16 neckPitchTarget;
+    /* 0x188 */ s16 neckRoll;
+    /* 0x18A */ s16 neckRollTarget;
```

Comment:
I don't think these are quite right. I've looked at it in kz and the first two definitely seem to be the neck's yaw:
https://github.com/user-attachments/assets/b2e635ff-31a4-4b50-8dca-67b4be6c2f02

I think this might be a world space vs. model space thing, because his head is essentially twisted 90 degrees in model space:
![image](https://github.com/user-attachments/assets/db9477be-042a-4c6d-82c8-d25d8d43977c)